libs/utils/utils.js

```diff
@@ -245,14 +245,20 @@ export const loadScript = (url, type) => new Promise((resolve, reject) => {
     if (type) {
       script.setAttribute('type', type);
     }
-    script.onload = () => {
-      resolve(script);
-    };
-    script.onerror = () => {
-      reject(new Error('error loading script'));
-    };
     head.append(script);
   }
+
+  if (script.dataset.loaded) {
```

Comment:
if `script` exists we should `resolve(script)` (line 241)

Reply:
In the case that script exists but is not loaded yet then the event listener will `resolve(script)` for us. If the script is already loaded then the event listener has already triggered so it needs to be resolved here. If we resolve before the script has loaded then the consumer of `loadScript()` will have to check and wait for the script to load.

Reply:
Thanks for the explanation - I mistakenly thought the `if (!script)` wrapped the entire block.

I think this will work great - two small changes noted in comments below